### ISSUE TITLE
Generify the javascript code that interacts with the workbench

### ISSUE THREAD

**Issue by wimjongman** [Tasktop/Foglyn#50]
Hi Peter,

In eclipse we have this bug open [1] to make a connection between a browser based application in a view and a normal view. Maybe you can give some pointers or want to comment on the bug.

[1] https://bugs.eclipse.org/bugs/show_bug.cgi?id=514294

**Comment by pstibrany** [comment]
Hi Wim,

I haven't done any more work in that direction in Foglyn, and truth to be told, I've switched to IntelliJ soon after, so there was no more Foglyn development at all.

I have tried to find the source code I've used for that video, but unfortunately I cannot see it anywhere on my computer or any of my old backups :-(

I remember that it worked by injecting little Javascript code into the web view to get notifications on page load (e.g. to supply login credentials when user reached the login page), when user tried to navigate to different case, or open attachment. I simply captured click events on links using jQuery and passed them to my Java code. When opening attachment, event from web view also included coordinates, so I was able to show Eclipse menu on right position on the screen.

It was all too hacky, relied on FogBugz web UI internals and on some internal Mylyn APIs, but I got that proof-of-concept working in just couple of hours and I thought that such integration between their UI and Eclipse features may be the way of moving Foglyn forward, instead of reimplementing UI myself using incomplete FogBugz API. (https://github.com/Tasktop/Foglyn/wiki/Foglyn2)

I don't think I have anything useful to add to https://bugs.eclipse.org/bugs/show_bug.cgi?id=514294, but thanks for reaching out to me. It's nice to see that my little proof-of-concept video from my Foglyn days attracted some interest after all ;-)
